### ISSUE TITLE
aws.cognito.UserPoolDomain changing certificateArn tries to replace

### ISSUE THREAD

**Issue by shellicar** [pulumi/pulumi-aws#998]
## Version
Windows - v2.3.0

## Description

If the certificate arn of a UserPoolDomain changes, the provider tries to replace the domain by deleting it and then recreating it.
This runs into all kind of issues, as even after deletion, the underlying (hidden) CloudFront that AWS use will still exist, preventing recreation of the UserPoolDomain until it has been destroyed.
This is also an issue when the provider tries to replace a certificate, and decides to delete the old one that is in use.

In the AWS console, the user pool domain's certificate can easily be changed without deleting and recreating the domain.

## Error

```
Diagnostics:
  aws:cognito:UserPoolDomain (domain):
    error: Error creating Cognito User Pool Domain: InvalidParameterException: Domain already exists.
```

**Comment by leezen** [comment]
This appears to be an issue upstream where the certificate is marked as `ForceNew` https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_cognito_user_pool_domain.go#L34

**Comment by shellicar** [comment]
Is there any work around for this?

**Comment by leezen** [comment]
Unfortunately, not that I know of. This will require modifying the dependency to support the correct flags.

**Comment by stack72** [comment]
Hi @shellicar 

This was released in pulumi-aws v2.11.0 today

Please let us know if there are nay issues

Paul

**Comment by shellicar** [comment]
@stack72 @leezen 
Thanks for that.
I've run through the same scenario twice and it seems to update the certificate arn on the UserPoolDomain without issue.
